### ISSUE TITLE
Preserve id and owner for Maputnik?

### ISSUE THREAD

**Issue by jirik** [klokantech/gl-style-package-spec#12]
Right now we are removing style's [id, owner, and some other properties](https://github.com/klokantech/gl-style-package-spec/blob/master/task/transform.js#L8) during deploy. Maputnik directly accesses deployed style-cdn.json where id and owner is missing, so it generates new random id for the style and does not set the owner.

On the other hand, if we preserve id and owner in `style-cdn.json`, Maputnik preserves it too.

So, shouldn't we preserve id and owner in deployed styles?
@klokan @lukasmartinelli

**Comment by lukasmartinelli** [comment]
The **id** is the only thing that identifies the style uniquely. The id is a must for Maputnik - if there exists none it regenerates a new unique one like you said. Owner does not have any usage actually outside of the Mapbox ecosystem - you are just able to edit it anyway.

I originally stripped these properties away so that there is no Mapbox user information in the style (the user owner id).

The modified timestamp is also set by Maputnik - however I should probably remove that from the Maputnik - it only makes working with git worse.

So to summarize my suggestion:

- Keep id (give a good one like `openmaptiles-osm-bright`)
- Still strip owner, modified, etc.

**Comment by jirik** [comment]
Thanks for your response, Lukas.

Just to organize my thoughts. Right know, in style's repo in master branch there is `style.json` suitable for Mapbox Studio. On every change, `style-cdn.json` in gh-pages is generated and it is the file suitable for Maputnik. If someone uses Maputnik to edit the style and wants to commit it back to the repo's master, he needs to do some changes to put it back into Mapbox-suitable form. It's not only id and owner, it's also source id, sprite URL and glyph URL. Among these, it seems only source id is crucial to Mapbox Studio (without Mapbox-suitable source id I am not able to upload the style back to Mapbox). We can make some script to automate this.

What about storing style.json in the repo's master in form more suitable for Maputnik instead of Mapbox? It would change things little bit, but if we want users to use Maputnik in the first place, it makes sense to me. We could generate something like `style-mb.json` in gh-pages branch to simplify upload to Mapbox. And we would need some script to transform Mapbox-like style into Maputnik-like style and to save some Mapbox-required properties like source id in metadata in case someone prefers Mapbox to edit the style.

**Comment by lukasmartinelli** [comment]
> What about storing style.json in the repo's master in form more suitable for Maputnik instead of Mapbox? It would change things little bit, but if we want users to use Maputnik in the first place, it makes sense to me. We could generate something like style-mb.json in gh-pages branch to simplify upload to Mapbox. And we would need some script to transform Mapbox-like style into Maputnik-like style and to save some Mapbox-required properties like source id in metadata in case someone prefers Mapbox to edit the style.

I would prefer that. Then we can also point the `style.json` to our OpenMapTiles CDN.
It is also easier for users to edit (it will simplify the documentation) since no longer uploading of MBTiles etc. is necessary.

**Comment by jirik** [comment]
OK, so ID is preserved in `style-cdn.json`.

Now it is ale possible to save `style.json` into master from either Mapbox or Maputnik.
If someone is using both editors for editing the same style, it is possible to put properties required by Mapbox (e.g. owner or source URL) into metadata. See https://github.com/openmaptiles/osm-bright-gl-style/blob/master/style.json#L38

In gh-pages is automatically generated also `style-mb.json` suitable for uploading to Mapbox. In this case, metadata mentioned above is used.
